File: 2020/04/22/leetcode-199/leetcode_199.go
package main

type TreeNode struct {
	Val int
	Left *TreeNode
	Right *TreeNode
}

func rightSideView(root *TreeNode) []int {
	result := make([]int, 0)
	if root == nil {
		return result
	}
	depthMap := map[int][]int{0: []int{root.Val}}

	stack := []*TreeNode{root}

	for {
		size := len(stack)
		if size == 0 {
			break
		}



	}


	return result
}

File: 2020/04/22/leetcode-199/leetcode_199.py
from collections import deque

class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def rightSideView(self, root: TreeNode) -> list:
       layer_dict = dict()
       max_depth = -1

       queue = deque([(root, 0)])

       while queue:
            node, depth = queue.popleft()
            if node is not None:
                max_depth = max(max_depth, depth)
                layer_dict[max_depth] = node.val

                queue.append((node.left, max_depth + 1))
                queue.append((node.right, max_depth + 1))

       return [layer_dict[depth] for depth in range(max_depth + 1)]




